+++ b/cmake/modules/FindCrossGUID.cmake	2020-04-23 20:12:14.165341492 +0000
@@ -46,10 +46,11 @@
   set(CROSSGUID_LIBRARIES ${CROSSGUID_LIBRARY})
   set(CROSSGUID_INCLUDE_DIRS ${CROSSGUID_INCLUDE_DIR})
 else()
-  find_path(CROSSGUID_INCLUDE_DIR NAMES guid.hpp guid.h)
+  find_path(CROSSGUID_INCLUDE_DIR NAMES Guid.hpp)
 
-  find_library(CROSSGUID_LIBRARY_RELEASE NAMES crossguid)
-  find_library(CROSSGUID_LIBRARY_DEBUG NAMES crossguidd)
+  find_library(CROSSGUID_LIBRARY_RELEASE NAMES xg)
+  find_library(CROSSGUID_LIBRARY_DEBUG NAMES xg)
+  set(CROSSGUID_DEFINITIONS -DHAVE_NEW_CROSSGUID)
 
   include(SelectLibraryConfigurations)
   select_library_configurations(CROSSGUID)

+++ b/xbmc/utils/StringUtils.cpp	2020-04-23 20:52:40.899776876 +0000
@@ -17,7 +17,7 @@
 //------------------------------------------------------------------------
 
 #ifdef HAVE_NEW_CROSSGUID
-#include <guid.hpp>
+#include <Guid.hpp>
 #else
 #include <guid.h>
 #endif
